libursa/Utils.cpp

```diff
@@ -175,6 +178,84 @@ void RunWriter::write(FileId next) {
     prev = next;
 }
 
+PosixRunWriter::~PosixRunWriter() {
+    if (!buffer_.empty()) {
+        // PosixRunWriter buffer not flushed, fatal error.
+        spdlog::error("PosixRunWriter not flushed before destructing");
+        std::terminate();
+    }
+}
```

Comment:
This behavior seems a little unintuitive. What maybe use RAII to ensure the buffer is flushed?

Reply:
It's more about defensive programming. Non-flushed runwriter probably means a bug. We're doing our own buffering here so it will mean that run data will end up interleaved with the footer or other weirdness like that. Right now I think the flush is not technically necessary :thinking:.

And terminate is a bit... unfortunate, yes. C++ complained when I tried to throw in destructor :thinking:

Reply:
(as per discussion, will be fixed in a separate PR)